lib/rules/disallow-space-after-keywords.js

```diff
@@ -51,7 +51,9 @@ module.exports.prototype = {
         );
 
         if (keywords === true) {
-            keywords = defaultKeywords;
+            keywords = defaultKeywords.filter(function(keyword) {
+                return ['function', 'return'].indexOf(keyword) === -1;
```

Comment:
I would say it's more complicated than this... some keywords require immediately following parentheses (`if`, `for`, `catch`, etc.), intervening whitespace of which can always be removed. Others (`return`, `throw`, `function`, etc.) require whitespace only when necessary to separate them from keywords/identifiers (so `return[]` and `function(){}` are fine but `thrownewError()` and `returnfoo` and `functionbar(){}` aren't). I think you want to leverage and maybe extend/expose more of https://github.com/hzoo/node-jscs/blob/1885/lib/token-categorizer.js , which also deals with such cases (cf. its top-level regular expressions). But you _definitely_ don't want to ignore these keywords outright.